hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala

```diff
@@ -478,10 +478,12 @@ object LowerMatrixIR {
         }
 
         loweredChild.mapGlobals('global.insertFields(colsField ->
-          aggOutsideTransformer(scanOutsideTransformer(irRange(0, 'global(colsField).len).map(idxSym ~> let(__cols_array = 'global(colsField), sa = '__cols_array(idxSym)) {
-            aggInsideTransformer(scanInsideTransformer(b0))
-          })))
-        ))
+          aggOutsideTransformer(scanOutsideTransformer(ToArray(StreamZip(
+            FastIndexedSeq(ToStream(GetField(Ref("global", loweredChild.typ.globalType), colsFieldName)), StreamIota(0, 0)),
```

Comment:
Is the second stream is meant to be 1, 2, 3? Why isn't step = 1?

It seems like the old code never had access to the index, so why bother to include it here at all?

Reply:
totally a bug, cause of the test failures. The index is used in aggregations/scans to join the column stream with the agg/scan results.